### PR TITLE
Support using _tidb_rowid when get chunks

### PR DESCRIPTION
**Pull request by leoppro**
using `_tidb_rowid`(if exist) to divide chunks, and then mydumper can dump data concurrently if upstream database is tidb

### CODE REVIEW

**Comment by GregoryIan** [comment]
@kennytm PTAL

File: mydumper.c [hunk 1]
```diff
@@ -1955,8 +1970,14 @@ GList * get_chunks_for_table(MYSQL *conn, char *database, char *table, struct co
 	if (!field) goto cleanup;
 
 	/* Get minimum/maximum */
-	mysql_query(conn, query=g_strdup_printf("SELECT %s MIN(`%s`),MAX(`%s`) FROM `%s`.`%s`", select_hint, field, field, database, table));
-	g_free(query);
+//	if (detected_server == SERVER_TYPE_TIDB && !strcmp(field,"_tidb_rowid")){
```
Comment (GregoryIan) [inline mydumper.c]: remove these useless comment code

File: mydumper.c [hunk 2]
```diff
@@ -1907,20 +1907,35 @@ GList * get_chunks_for_table(MYSQL *conn, char *database, char *table, struct co
 	MYSQL_ROW row;
 	char *field = NULL;
 	int showed_nulls=0;
+    gchar *query = NULL;
+
+    /* If detected server is TiDB, try using _tidb_rowid*/
+    if(detected_server == SERVER_TYPE_TIDB){
```
Comment (GregoryIan) [inline mydumper.c]: Fix the code alignment

**Comment by GregoryIan** [comment]
Rest LGTM

**Comment by GregoryIan** [comment]
LGTM now

File: mydumper.c [hunk 3]
```diff
@@ -1907,18 +1907,26 @@ GList * get_chunks_for_table(MYSQL *conn, char *database, char *table, struct co
...
+	if (detect_tidb_rowid(conn,database,table)){
+		field = g_strdup("_tidb_rowid");
+	}
 
-	while ((row=mysql_fetch_row(indexes))) {
-		if (!strcmp(row[2],"PRIMARY") && (!strcmp(row[3],"1"))) {
-			/* Pick first column in PK, cardinality doesn't matter */
-			field=row[4];
-			break;
+    /* pick index, in future should be able to preset in configuration too */
```
Comment (leoppro) [inline mydumper.c]:
```suggestion
	/* pick index, in future should be able to preset in configuration too */
```

**Pull request merged**